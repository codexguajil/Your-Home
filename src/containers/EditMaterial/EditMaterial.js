import React from 'react';
import { Button, Modal, Form, Input } from 'antd';

export const CollectionCreateForm = Form.create({ name: 'form_in_modal' })(
  class extends React.Component {
    render() {
      const { visible, onCancel, onCreate, form, editMaterial } = this.props;
      const { getFieldDecorator } = form;
      return (
        <Modal
          visible={visible}
          title="Edit Material"
          okText="Update"
          onCancel={onCancel}
          onOk={onCreate}
        >
          <Form layout="vertical">
            <Form.Item label="Name">
              {getFieldDecorator('name', {
                rules: [{ required: false }],
              })(<Input onChange={editMaterial} name='name' placeholder='Update here, or leave blank to keep unchanged' />)}
            </Form.Item>
            <Form.Item label="Brand">
              {getFieldDecorator('brand', {
                rules: [{ required: false }],
              })(<Input onChange={editMaterial} name='brand' placeholder='Update here, or leave blank to keep unchanged' />)}
            </Form.Item>
            <Form.Item label="Model">
              {getFieldDecorator('model', {
                rules: [{ required: false }],
              })(<Input onChange={editMaterial} name='model'  placeholder='Update here, or leave blank to keep unchanged' />)}
            </Form.Item>
            <Form.Item label="Vendor">
              {getFieldDecorator('vendor', {
                rules: [{ required: false }],
              })(<Input onChange={editMaterial} name='vendor'  placeholder='Update here, or leave blank to keep unchanged' />)}
            </Form.Item>
            <Form.Item label="Quantity">
              {getFieldDecorator('quantity', {
                rules: [{ required: false }],
              })(<Input onChange={editMaterial} name='quantity'  placeholder='Update here, or leave blank to keep unchanged' />)}
            </Form.Item>
            <Form.Item label="Price">
              {getFieldDecorator('price', {
                rules: [{ required: false }],
              })(<Input onChange={editMaterial} name='price'  placeholder='Update here, or leave blank to keep unchanged' />)}
            </Form.Item>
            <Form.Item label="Manual URL">
              {getFieldDecorator('manual', {
                rules: [{ required: false }],
              })(<Input onChange={editMaterial} name='manual'  placeholder='Update here, or leave blank to keep unchanged' />)}
            </Form.Item>
            <Form.Item label="Notes">
              {getFieldDecorator('notes')(<Input onChange={editMaterial} name='notes' type="textarea" placeholder='Update here, or leave blank to keep unchanged' />)}
            </Form.Item>
            <Form.Item className="collection-create-form_last-form-item">
            </Form.Item>
          </Form>
        </Modal>
      );
    }
  }
);

class EditMaterial extends React.Component {
  state = { 
    visible: false,
    name: '',
    brand: '',
    vendor: '',
    model: '',
    quantity: '',
    price: '',
    manual: '',
    notes: ''
  };

  showModal = () => {
    this.setState({ visible: true });
  }

  handleCancel = () => {
    this.setState({ visible: false });
  }

  editMaterial = (e) => {
    this.setState({[e.target.name]: e.target.value});
  }

  handleCreate = () => {
    const { name, brand, vendor, model, quantity, price, manual, notes } = this.state;
    const updatedMaterials = { name, brand, vendor, model, quantity, price, manual, notes };
    console.log(updatedMaterials);
    this.setState({ visible: false});
  }

  render() {
    return (
      <div className="edit-material-btn">
        <Button onClick={this.showModal} size="small" type="link">
          <i className="fas fa-pen"></i>
        </Button>
        <CollectionCreateForm
          visible={this.state.visible}
          onCancel={this.handleCancel}
          onCreate={this.handleCreate}
          editMaterial={this.editMaterial}
        />
      </div>
    );
  }
}

export default EditMaterial;