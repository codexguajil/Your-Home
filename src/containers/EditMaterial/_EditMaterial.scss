.sub-title {
  margin-bottom: 0;
}