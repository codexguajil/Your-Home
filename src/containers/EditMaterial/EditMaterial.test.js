import React from 'react';
import { shallow } from 'enzyme';
import { EditMaterial } from './EditMaterial';
import { CollectionCreateForm } from './EditMaterial';

describe('EditMaterial', () => {
  let wrapper;
  let formWrapper;
  let mockFn;

  beforeEach(() => {
    mockFn = jest.fn();
    wrapper = shallow(<EditMaterial/>)
    formWrapper = shallow(
      <CollectionCreateForm 
        visible={mockFn}
        onCancel={mockFn}
        onCreate={mockFn}
        form={mockFn}
        editMaterial={mockFn}
    />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
    expect(formWrapper).toMatchSnapshot();
  });

  it('should update state visible when showModal is called', () => {
    wrapper.setState({ visible: false })
    wrapper.instance().showModal();
    expect(wrapper.state('visible')).toBe(true);
  });

  it('should update state visible when handleCancel is called', () => {
    wrapper.setState({ visible: true })
    wrapper.instance().handleCancel();
    expect(wrapper.state('visible')).toBe(false);
  });

  it('should updateState for correct name and value on editMaterial', () => {
    const mockedEvent = { target: {name: 'name', value: 'new name'} };
    wrapper.instance().editMaterial(mockedEvent);
    expect(wrapper.state('name')).toBe('new name');
  });

  it('should should update state visible', () => {
    wrapper.setState({ visible: true })
    wrapper.instance().handleCreate();
    expect(wrapper.state('visible')).toBe(false);
  });
});