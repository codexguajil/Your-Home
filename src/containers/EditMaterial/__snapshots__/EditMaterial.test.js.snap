// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditMaterial should match snapshot 1`] = `
<div
  className="edit-material-btn"
>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    size="small"
    type="link"
  >
    <i
      className="fas fa-pen"
    />
  </Button>
  <Form(Component)
    onCancel={[Function]}
    onCreate={[Function]}
    visible={false}
    wrappedComponentRef={[Function]}
  />
</div>
`;
